File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    
    ans=0
    def dfs(self,root,total):
        
        if not root:
                
                return
        
        if not root.left and not root.right:
                total=total*10+root.val
                self.ans +=total
                return
            
        total=total*10+root.val
        self.dfs(root.left,total)
        self.dfs(root.right,total)
            
        
    
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        
            total=0
            self.dfs(root,total)
            return self.ans
        

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    int total=0;
    void totalBinary(TreeNode* node,int val){
        
            if(!node) return;
        
            if(node->left==NULL and node->right==NULL){
                    total+=(val*10+node->val);
                    return;            
            }
            
            val*=10;
            totalBinary(node->left,val+node->val);
            totalBinary(node->right,val+node->val);   
    }
    
    
    int sumNumbers(TreeNode* root) {
        
            if(root==NULL){
                    return 0;
            }

        
            totalBinary(root,0); 
            return total;
        
    }
};